src/web3Provider/web3VmProvider.js

```diff
@@ -212,8 +212,13 @@ function getSha3Input (stack, memory) {
   memoryStart = parseInt(memStartDec) * 2
   var memLengthDec = (new ethutil.BN(memoryLength.replace('0x', ''), 16).toString(10))
   memoryLength = parseInt(memLengthDec) * 2
-  var memoryHex = memory.join('')
-  var sha3Input = memoryHex.substr(memoryStart, memoryLength)
+
+  var subMemoryIndex = Math.floor(memoryStart / 32)
+  var sha3Input = ''
+  while (sha3Input.length < memoryLength) {
+    sha3Input += memory[subMemoryIndex]
```

Comment:
Ah, I think I misunderstood your question earlier. Unfortunately this won't work. Types in memory most of the time have a length of 32 bytes but
1) they might be offset by some amount
2) this does not apply to sha3